ner4opt/ner4opt.py

```diff
@@ -0,0 +1,192 @@
+                  Options include `lexical`, `lexical_plus`, `semantic` and `hybrid`
+                  Default is `hybrid`
+    use_gpu (bool) : Specifies the model to use gpu while calculating transformer based features.
+                     Default is False
+
+    Methods
+    -------
+    get_entities()
+
+    Attributes
+    -------
+    text (str) : Sample optimization problem
+
+    Returns
+    -------
+
+    Returns named entities of a given optimization problem. A list of dictionaries for each entity
+
+    Keys:
+        Name: start, dtype: int: Starting character index of the entity. Range is [0, len(text)]
+        Name: end, dtype: int: Ending character index of the entity. Range is [0, len(text)]
+        Name: word, dtype: str: Entity phrase
+        Name: entity_group, dtype: str: Type of the entity
+        Name: score, dtype: float: Defines the confidence of the prediction. Range is [0.0, 100.0]
+
+    Example Usage
+    -------------
+    # Import the Ner4Opt Library
+    >>> from ner4opt import Ner4Opt
+
+    # Problem Description
+    >>> problem_description = "Cautious Asset Investment has a total of $ 150,000 to manage and decides to invest it in money market fund , which yields a 2 % return as well as in foreign bonds , which gives and average rate of return of 10.2 % . Internal policies require PAI to diversify the asset allocation so that the minimum investment in money market fund is 40 % of the total investment . Due to the risk of default of foreign countries , no more than 40 % of the total investment should be allocated to foreign bonds . How much should the Cautious Asset Investment allocate in each asset so as to maximize its average return ?"
+
+    # Ner4Opt Model options: lexical, lexical_plus, semantic, hybrid
+    >>> ner4opt = Ner4Opt(model='hybrid')
+
+    # Extracts a list of dictionaries corresponding to each entity in the problem description
+    # Each dictionary holds keys for
+    # start (starting character index of the entity),
+    # end (ending character index of the entity),
+    # word (entity),
+    # entity_group (entity label) and
+    # score (confidence score for the entity)
+    >>> entities = ner4opt.get_entities(problem_description)
+
+    # Output
+    >>> print(entities)
+
+    # Displaying a prettyprint of few entities for understanding
+    [
+        {
+            ...
+        },
+
+        {
+            'start': 32,
+            'end': 37,
+            'word': 'total',
+            'entity_group': 'CONST_DIR',
+            'score': 0.997172257043559
+        },
+        {
+            'start': 575,
+            'end': 583,
+            'word': 'maximize',
+            'entity_group': 'OBJ_DIR',
+            'score': 0.9982091561140413
+        },
+        {
+            ...
+        },
+
+    ]
+
+    Help
+    ----
+    >>> from ner4opt import Ner4Opt
+    >>> Ner4Opt.__doc__
+    """
+
+    def __init__(self, model: str = Constants.HYBRID, use_gpu: bool = False):
```

Comment:
The design/naming of _crf_model and _roberta_model has a subtle misleading effect. 
First, they notion of hybrid is now lost, and it is called like roberta model
Second, It ALMOST look like we are not using CRF in the hybrid model which is not true. 

How about the following design: 

```
      # Load model
        if self._model == Constants.LEXICAL:
            self._lexical_model = joblib.load(self._lexical_model_path)

        elif self._model == Constants.LEXICAL_PLUS:
            self._lexical_model = joblib.load(self._lexical_plus_model_path)

        elif self._model == Constants.SEMANTIC:
            self._semantic_model = load_torch_model(Constants.SEMANTIC_MODEL_ROBERTA_V1, use_gpu=use_gpu)

        elif self._model == Constants.HYBRID:
            self._lexical_model = joblib.load(self._hybrid_model_path)
            self._semantic_model = load_torch_model(Constants.SEMANTIC_MODEL_ROBERTA_V2, use_gpu=use_gpu)
```

Notice the replacement of crf_model with lexical_model and roberta_model with semantic model
Notice also the swap of lines under Hybrid when loading to make it consistent

Reply:
Our (purely) semantic model is the odd one out, which causes the confusion. It relies on the probabilities of entity label predictions and doesn't use CRF as the final layer. All the other models use CRF as the final layer for entity label predictions, with different types of features as inputs (lexical features, lexical features + lexical domain specific features, lexical features +  lexical domain specific features + semantic features). I changed the naming to call all CRF models as _crf_model and deep models as just _semantic_feature_extractor. Also added more comments. Hope that is more clear, please let me know otherwise.